src/main.rs

```diff
@@ -188,6 +197,92 @@ fn main() {
         process::exit(0);
     }
 
+    if argv[1] == "try-mine" {
+        if argv.len() < 3 {
+            eprintln!(
+                "Usage: {} try-mine <working-dir>
+
+Given a <working-dir>, try to ''mine'' an anchored block. This invokes the miner block
+assembly, but does not attempt to broadcast a block commit. This is useful for determining
+what transactions a given chain state would include in an anchor block, or otherwise
+simulating a miner.
+",
+                argv[0]
+            );
+            process::exit(1);
+        }
+
+        let sort_db_path = format!("{}/burnchain/db/bitcoin/mainnet/sortition.db", &argv[2]);
+        let chain_state_path = format!("{}/chainstate/", &argv[2]);
+
+        let sort_db = SortitionDB::open(&sort_db_path, false)
+            .expect(&format!("Failed to open {}", &sort_db_path));
+        let chain_id = core::CHAIN_ID_MAINNET;
+        let (chain_state, _) = StacksChainState::open(true, chain_id, &chain_state_path)
+            .expect("Failed to open stacks chain state");
+        let chain_tip = SortitionDB::get_canonical_burn_chain_tip(sort_db.conn())
+            .expect("Failed to get sortition chain tip");
+
+        let mempool_db =
+            MemPoolDB::open(true, chain_id, &chain_state_path).expect("Failed to open mempool db");
+
+        let stacks_block = chain_state.get_stacks_chain_tip(&sort_db).unwrap().unwrap();
+        let parent_header = StacksChainState::get_anchored_block_header_info(
+            chain_state.db(),
+            &stacks_block.consensus_hash,
+            &stacks_block.anchored_block_hash,
+        )
+        .expect("Failed to load chain tip header info")
+        .expect("Failed to load chain tip header info");
+
+        let sk = StacksPrivateKey::new();
+        let mut tx_auth = TransactionAuth::from_p2pkh(&sk).unwrap();
+        tx_auth.set_origin_nonce(0);
+
+        let mut coinbase_tx = StacksTransaction::new(
+            TransactionVersion::Mainnet,
+            tx_auth,
+            TransactionPayload::Coinbase(CoinbasePayload([0u8; 32])),
+        );
+
+        coinbase_tx.chain_id = chain_id;
+        coinbase_tx.anchor_mode = TransactionAnchorMode::OnChainOnly;
+        let mut tx_signer = StacksTransactionSigner::new(&coinbase_tx);
+        tx_signer.sign_origin(&sk).unwrap();
+        let coinbase_tx = tx_signer.get_tx().unwrap();
+
+        let mainnet_block_limit = ExecutionCost {
```

Comment:
Isn't this a constant somewhere?

Reply:
Yeah, in `stacks-node::config` unfortunately -- I can refactor it to be `core::` constant, though.